lib/SetAPI/generic/SetInterfaceV1.py

```diff
@@ -89,12 +131,82 @@ def _check_save_set_params(self: "SetInterfaceV1", params: dict[str, Any]) -> No
             if not params.get(param):
                 raise ValueError(param_required(param))
 
+        # this should not happen with the current API, but best to check
+        if not set_item_name or set_item_name not in SET_ITEM_NAME_TO_SET_TYPE:
+            err_msg = f"invalid set item name: {set_item_name}"
+            raise ValueError(err_msg)
+
+        # bail out early for RNASeqSampleSets
+        if set_item_name == RNASEQ_SAMPLE:
+            return set_item_name
+
+        # convert the set item name into the set type
```

Comment:
the checks below were carried out by the individual classes; this combines them, with the option of skipping the checks that don't apply to sample sets of certain types